Mage/src/main/java/mage/watchers/common/FirstSpellCastThisTurnWatcher.java

```diff
@@ -58,10 +58,6 @@ public void reset() {
     }
 
     public UUID getIdOfFirstCastSpell(UUID playerId) {
-        if (playerFirstSpellCast.get(playerId) == null) {
-            return playerFirstCastSpell.get(playerId);
-        } else {
-            return playerFirstSpellCast.get(playerId);
-        }
+        return playerFirstSpellCast.getOrDefault(playerId, playerFirstSpellCast.get(playerId));
```

Comment:
This code block was pretty weird. We are trying to get value from map, and if map does not contain value it still tryies to get value?
Maybe 
`return playerFirstSpellCast.getOrDefault(playerId, null);`
is the correct option?